api/main.go

```diff
@@ -49,40 +49,43 @@ func main() {
 	app.Version = Version
 	app.Flags = config.APIFlags()
 	app.Action = func(c *cli.Context) error {
-		cfg := config.NewContextAPIConfig(c)
-		if err := cfg.Validate(); err != nil {
+		if err := config.ValidateAPIContext(c); err != nil {
 			return err
 		}
 
 		logger := logging.NewLogWriter(c.Bool("debug"))
 		log.SetOutput(logger)
 
 		awsConfig := defaults.Get().Config
-		staticCreds := credentials.NewStaticCredentials(cfg.AccessKey(), cfg.SecretKey(), "")
+		staticCreds := credentials.NewStaticCredentials(
+			c.String(config.FlagAWSAccessKey.GetName()),
+			c.String(config.FlagAWSSecretKey.GetName()),
+			"")
+
 		awsConfig.WithCredentials(staticCreds)
-		awsConfig.WithRegion(cfg.Region())
-		awsConfig.WithMaxRetries(cfg.MaxRetries())
+		awsConfig.WithRegion(c.String(config.FlagAWSRegion.GetName()))
+		awsConfig.WithMaxRetries(config.DefaultMaxRetries)
 		session := session.New(awsConfig)
 
-		delay := c.Duration(config.FLAG_AWS_TIME_BETWEEN_REQUESTS)
+		delay := c.Duration(config.FlagAWSRequestDelay.GetName())
 		ticker := time.Tick(delay)
 		session.Handlers.Send.PushBack(func(r *request.Request) {
 			<-ticker
 		})
 
 		client := awsclient.NewClient(session)
-		tagStore := tag.NewDynamoStore(session, cfg.DynamoTagTable())
-		jobStore := job.NewDynamoStore(session, cfg.DynamoJobTable())
+		tagStore := tag.NewDynamoStore(session, c.String(config.FlagAWSTagTable.GetName()))
```

Comment:
similar to my last comment re: simplifying getting such values, but it seems like having to know that `config.FlagAWSTagTable.GetName()` must be converted by `c.String()` is awkward

Reply:
see above